crates/esthri-cli/src/main.rs

```diff
@@ -582,10 +566,19 @@ async fn async_main() -> Result<()> {
     let credentials_provider = DefaultCredentialsProvider::new().unwrap();
     let s3 = S3Client::new_with(http_client, credentials_provider, Region::default());
 
-    match cli {
-        Cli::Esthri(esthri_cli) => dispatch_esthri_cli(esthri_cli.cmd, &s3).await?,
-        Cli::AwsCompat(aws_cli) => dispatch_aws_cli(aws_cli.cmd, &s3).await?,
-    }
+    match aws_compat_mode {
```

Comment:
Just use an if statement here

Reply:
> the use of `impl AsRef<str>` was an intentional design choice, not really understanding the rationale for changing things back to plain references

@silverjam 

String already implements AsRef<str> so we can use it freely at call site where both String and &str could be taken, if using String then the caller can simply take a reference, if &str then the function signature is simple instead of using the redundant `as_ref()` which is littered through the code base, like: https://github.com/swift-nav/esthri/blob/d1dc9d9959a7a6a80f61ead288dd449daab50de1/crates/esthri/src/ops/delete.rs#L107
We also have cases where ownership of the str must be taken, and we have it covered with copy-on-write wrapper, like: https://github.com/swift-nav/esthri/blob/03fe7088ad58f9c7d22bce8046f408e24d16ce43/crates/esthri/src/ops/upload.rs#L569, so no issues there

I think earlier code used `impl AsRef<str>` a lot but more recent code started using `&str`, because most of the helper functions that do the heavy lifting take &str anyway, it makes everything simpler. The entire sync.rs module currently only uses &str for example.

I see lot of project getting away from the `impl AsRef<str>` pattern due to some sizedness constraint on some data structures, &str is supposedly more idiomatic. Please correct me if I am wrong on some accounts 😄